src/sortStudents.ts

```diff
@@ -1,16 +1,58 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+// function for average
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+function avGrad(grades: number[]): number {
+  const av = grades.reduce((sum, element) => sum + element, 0) / grades.length;
+
+  return av;
+}
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const copiedArray = [...students];
+
+  copiedArray.sort((a, b) => {
+    switch (sortBy) {
+      case SortType.Name:
+      case SortType.Surname:
+        return (order === 'asc' ? 1 : -1) * a[sortBy].localeCompare(b[sortBy]);
+
+      case SortType.Age:
+      case SortType.Married:
+        return (order === 'asc' ? 1 : -1) * (+a[sortBy] - +b[sortBy]);
+
+      case SortType.AverageGrade:
+        return (
+          (order === 'asc' ? 1 : -1)
+        * (avGrad(a.grades) - avGrad(b.grades))
```

Comment:
```suggestion
          (order === 'asc' ? 1 : -1)
            * (avGrad(a.grades) - avGrad(b.grades))
```
I see this type of indentation in the case of multiline operations more often. In my opinion, it's more readable.